features/guidance/staggered-intersections.feature

```diff
@@ -98,3 +98,48 @@ Feature: Staggered Intersections
             | waypoints | route                         | turns                              |
             | a,g       | Oak St,Cedar Dr,Elm St,Elm St | depart,turn right,turn left,arrive |
             | g,a       | Elm St,Cedar Dr,Oak St,Oak St | depart,turn right,turn left,arrive |
+
+    Scenario: Staggered Intersection: do not collapse if a mode change is involved
+        Given the node map
+            """
+                j
+            a b c
+                d
+                e  f g
+                h
+            """
+
+        And the ways
+            | nodes  | highway | name     | route |
+            | abc    | primary | to_sea   |       |
+            | ef     |         | to_sea   | ferry |
+            | fg     | primary | road     |       |
+            | jcdeh  | primary | road     |       |
+
+        When I route I should get
+            | waypoints | route                          | turns                              | modes                         |
+            | a,g       | to_sea,road,to_sea,road,road   | depart,turn right,turn left,notification straight,arrive | driving,driving,ferry,driving,driving |
+            | g,a       | road,to_sea,road,to_sea,to_sea | depart,notification straight,turn right,turn left,arrive | driving,ferry,driving,driving,driving |
+
+    Scenario: Staggered Intersection: do not collapse intermediary intersections
+        Given the node map
+            """
+                j
+            a b c
+                e f g
+                k l m
+                i
+            """
```

Comment:
Hm also what I just realized, think about this scenario

```
    d e f
a b c
    g h i
```

and a route a,g.

Because we're only working on RouteSteps in post-processing we're detecting a Staggered Intersection here and collapse it away emitting nothing at c. There is no way to get information about the second ambiguous way cdef.

To handle this correctly Staggered Intersections have to be detected in pre-processing and be a separate (internal) instruction. Thoughts @MoKob?